### DISCUSSION TITLE
What is the difference between retinanet and rotate-retinanet?

### DISCUSSION THREAD

**Discussion by vansin** [open-mmlab/mmrotate#691] [Q&A] [answered]
如何理解 retinanet 和 rotate-retinanet 间的区别？
What is the difference between retinanet and rotate-retinanet?

是不是只有Head上有不同之处，Head上不同之处具体是什么？是否能用模型图来表示？
换句话说是在mmdet的经典模型上，需要改什么能够让原算法支持旋转目标检测~

**Comment by vansin** [comment]
貌似只有检测头是不一样的~

# mmdet 中 retinanet

![image](https://raw.githubusercontent.com/vansin/mmgraph/83bbdbe74267f0212682fd448a3f0218b68e181c/mmdetection/configs/retinanet/retinanet_r50_fpn_1x_coco.py.svg)


# mmrotate 中的 retinanet


![image](https://raw.githubusercontent.com/vansin/mmgraph/83bbdbe74267f0212682fd448a3f0218b68e181c/mmrotate/configs/rotated_retinanet/rotated-retinanet-rbox-le90_r50_fpn_1x_dota.py.svg)

**Comment by yangxue0827** [comment, marked as answer]
将水平框检测代码改成旋转框最简单的方式就是在head处增加一个角度参数的估计，其他模型结构可以保持不变。只是有些非模型结构的操作也需要做相应的修改，如anchor（不必要，现在主流还是用水平anchor避免引入过多旋转anchor，以减少计算量增加召回率，详细可以看R3Det文章对水平/旋转anchor的比较分析）、IoU的计算（涉及nms、label assign等）、框的编解码器、可视化。
